crates/matrix-sdk/src/sliding_sync/list/builder.rs

```diff
@@ -29,6 +34,22 @@ pub struct SlidingSyncListBuilder {
     timeline_limit: Option<UInt>,
     name: String,
     ranges: Vec<(UInt, UInt)>,
+
+    /// Should this list be cached and reloaded from the cache?
+    cache_policy: SlidingSyncListCachePolicy,
+
+    /// Total number of rooms that is possible to interact with the given list.
+    /// See also comment of [`SlidingSyncList::maximum_number_of_rooms`].
+    /// May be reloaded from the cache.
```

Comment:
The only way to set those values are when it's reloaded from the cache. The doc' should emphasize that :-). It's not “may be reloaded from the cache” but rather “contains the default value, or the value from the cache, cannot be modified by the user”.

Question: Did we read those values from the cache previously?

Reply:
Yes, we did! That was the data that was merged in `set_from_cold`, using observables (while in theory we should only have to use plain assignments, assuming reading from cache only happens at initialization).

Reply:
And good point; I've actually tweaked the data stored here, by introducing a new temporary data structure that contains only the cached data. That way, we can decide in `build()` if we read the cached data or the default, and that puts the code assigning the defaults closer to the data read from the cache.